src/popup.js

```diff
@@ -0,0 +1,157 @@
+/*global chrome */
+
+import { Status } from "./status.js";
+
+const ids = {
+    status: "dgtw-status",
+    restore: "dgtw-restore",
+    showChanges: "show-changes",
+    showChangesCheckbox: "show-changes-checkbox",
+    showHighlights: "show-highlights",
+    showHighlightsCheckbox: "show-highlights-checkbox"
+};
+
+// Controls the DGtW content script on the current page
+function sendMessageToContentScript(type, callback) {
+    // If this is a test, the popup is opened in a tab. This steals focus from
+    // the tab in which the content script is running.
+    const queryInfo = { active: !isTest, currentWindow: true };
+
+    chrome.tabs.query(queryInfo, function(tabs) {
+        // Normally, only one tab matches the query.
+        // When testing, a blank tab might match the query. Using `length - 1`
+        // picks the correct target, the most recently opened background tab.
+        const targetTab = tabs[tabs.length - 1].id;
+        chrome.tabs.sendMessage(targetTab, { type: type }, callback);
+    });
+}
+
+function showElement(id) {
+    document.getElementById(id).classList.remove("hide");
+    document.getElementById(id).classList.add("show");
+}
+
+function hideElement(id) {
+    document.getElementById(id).classList.remove("show");
+    document.getElementById(id).classList.add("hide");
+}
+
+function setStatusTo(newStatus) {
+    let statusText = "<i>Degender the web</i> ";
+    switch (newStatus) {
+        case Status.excludedDomain:
+            statusText += "does not run on this site due to ";
+            statusText += "technical incompatibility.";
+
+            // Show no buttons
+            hideElement(ids.showChanges);
+            hideElement(ids.showHighlights);
+            hideElement(ids.restore);
```

Comment:
Yeah, I see the repetition below. I could imagine adding a function like this:
<pre>
setElementVisibility(showChanges:bool, showHighlights:bool, restore:bool):void
</pre>
That would be a little more concise. But it would also be a little less self-explanatory, since the settings for the three elements would be identified by argument position rather than name. What do you think about the tradeoff?

Reply:
I agree. Thanks for raising the issue.